src/parquet/arrow/reader.cc

```diff
@@ -1159,5 +1328,102 @@ Status ColumnReader::NextBatch(int batch_size, std::shared_ptr<Array>* out) {
   return impl_->NextBatch(batch_size, out);
 }
 
+// StructImpl methods
+
+Status StructImpl::CreateBitmap(size_t num_elements,
+    std::shared_ptr<PoolBuffer>* null_bitmap_out) {
+  *null_bitmap_out = std::make_shared<::arrow::PoolBuffer>(pool_);
+  auto bitmap_size = ::arrow::BitUtil::CeilByte(num_elements) / 8;
+  RETURN_NOT_OK((*null_bitmap_out)->Resize(bitmap_size));
+  uint8_t* null_bitmap_ptr = (*null_bitmap_out)->mutable_data();
+  memset(null_bitmap_ptr, 0xFF, bitmap_size);
```

Comment:
The fact that its zero'd is not a real problem, however there's the issue of getting a `MutableBuffer` instead of a `PoolBuffer`. Is there any downside of using the first kind in this context instead of the other?